devito/types/equation.py

```diff
@@ -81,11 +82,14 @@ def evaluate(self):
 
         The RHS of the Equation is evaluated at the indices of the LHS if required.
         """
-        try:
-            lhs, rhs = self.lhs.evaluate, self.rhs._eval_at(self.lhs).evaluate
-        except AttributeError:
-            lhs, rhs = self._evaluate_args()
-        eq = self.func(lhs, rhs, subdomain=self.subdomain,
+        if self.is_Break:
+            args = self._evaluate_args()
```

Comment:
Not sure why Break would be a property of Eq.

Reply:
"Break" is a type of equation, just like "Increment". There is a property "is_Increment" and so there should be a property "is_Break", shouldn't it?

All expressions passed to the "Operator" class must have some properties exclusive to the "Eq" type, such as "lhs" and "lhs.function", which are used in various places. That is why I had to inherit from it when creating the "Break" class.

Reply:
I think this:

> "Break" is a type of equation

is what @mloubout was arguing about